test/helpers/asl-db.js

```diff
@@ -1,41 +1,17 @@
 const Schema = require('@asl/schema');
+const snakeCase = str => str.replace(/[A-Z]/g, s => `_${s.toLowerCase()}`);
 
 module.exports = settings => {
 
   return {
     init: (populate, keepAlive) => {
       const schema = Schema(settings);
-      const tables = [
-        'Notification',
-        'TrainingPil',
-        'TrainingCourse',
-        'AsruEstablishment',
-        'Procedure',
-        'Rop',
-        'ProjectVersion',
-        'Project',
-        'Permission',
-        'Authorisation',
-        'PilTransfer',
-        'PIL',
-        'PlaceRole',
-        'Place',
-        'Role',
-        'Exemption',
-        'Certificate',
-        'EmailPreferences',
-        'Profile',
-        'Invitation',
-        'Establishment'
-      ];
-      return tables.reduce((p, table) => {
-        return p.then(() => {
-          if (typeof schema[table].queryWithDeleted === 'function') {
-            return schema[table].queryWithDeleted().hardDelete();
-          }
-          return schema[table].query().truncate();
-        });
-      }, Promise.resolve())
+
+      const tableNames = Object.keys(schema)
+        .filter(key => !!schema[key].tableName)
+        .map(key => snakeCase(schema[key].tableName));
+
+      return schema.knex.raw(`TRUNCATE TABLE ${tableNames.join(', ')} CASCADE;`)
```

Comment:
Running a `TRUNCATE TABLE x CASCADE;` query per table in series appears to be considerably slower than just deleting the rows.

However, what if we just have a single SQL statement that truncates all the tables at once? 

If this is also slow then there might be the option to temporarily disable the FK constraints with 

```
ALTER TABLE x DISABLE TRIGGER ALL;
DELETE FROM x;
ALTER TABLE x ENABLE TRIGGER ALL;
```